### PR TITLE
feat: Envelope

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: Sources/Sentry/SentryEnvelope.m [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#if __has_include(<Sentry/Sentry.h>)
+//#import <Sentry/SentryEvent.h>
+//#import <Sentry/SentryEnvelope.h>
```
Comment (bruno-garcia) [inline Sources/Sentry/SentryEnvelope.m]: Need to sort these out

File: Sources/Sentry/include/Sentry.h [hunk 2]
```diff
@@ -16,6 +16,7 @@ FOUNDATION_EXPORT const unsigned char SentryVersionString[];
 
 #if __has_include(<Sentry/Sentry.h>)
 
+//#import <Sentry/SentryEnvelope.h>
```
Comment (bruno-garcia) [inline Sources/Sentry/include/Sentry.h]: same

**Comment by HazAT** [comment]
To give you more context why we have `#if __has_include(<Sentry/Sentry.h>)` all over the place and which are super shitty to maintain.
We have a static lib target and back then when I introduced this, this was to only way to make Framework and static lib work at the same time.
We can try to revisit this, 2 reasons:
First, we never upload the static lib anywhere, we only publish the Carthage build and Cocoapods.
Secondly, it might not even be necessary.

File: Sources/Sentry/SentryEnvelope.m [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- (instancetype)initWithHeader:(SentryEnvelopeHeader *)header
+                    singleItem:(SentryEnvelopeItem *)item {
+    return [self initWithHeader:header items:@[item]];
+}
+
+- (instancetype)initWithHeader:(SentryEnvelopeHeader *)header
+                         items:(NSArray<SentryEnvelopeItem *> *)items {
+    if (self = [super init]) {
+        self->_header = header;
+        self->_items = items;
```
Comment (HazAT) [inline Sources/Sentry/SentryEnvelope.m]:
```suggestion
        _header = header;
        _items = items;
```
There is no need for `self->`

File: Sources/Sentry/SentryEnvelope.m [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+@end
+
+@implementation SentryEnvelopeItem
+
+- (instancetype)initWithHeader:(SentryEnvelopeItemHeader *)header
+                          data:(NSData *)data {
+    if (self = [super init]) {
+        self->_header = header;
+        self->_data = data;
```
Comment (HazAT) [inline Sources/Sentry/SentryEnvelope.m]:
```suggestion
        _header = header;
        _data = data;
```

File: Sources/Sentry/SentryEnvelope.m [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+#import "SentryEnvelope.h"
+#endif
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryEnvelopeHeader
+
+- (instancetype)initWithId:(NSString *) envelopeId {
+    if (self = [super init]) {
+        self->_envelopeId = envelopeId;
```
Comment (HazAT) [inline Sources/Sentry/SentryEnvelope.m]:
```suggestion
        _envelopeId = envelopeId;
```

File: Sources/Sentry/SentryEnvelope.m [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+@end
+
+@implementation SentryEnvelopeItemHeader
+
+- (instancetype)initWithType:(NSString *)type
+                length:(NSUInteger)length {
+    if (self = [super init]) {
+        self->_type = type;
+        self->_length = length;
```
Comment (HazAT) [inline Sources/Sentry/SentryEnvelope.m]:
```suggestion
        _type = type;
        _length = length;
```

File: Sources/Sentry/SentryEnvelope.m (hunk 3 above)
Comment (bruno-garcia) [reply Sources/Sentry/SentryEnvelope.m]: Doesn't build. For some reason can't access the field. Will try again

**Pull request merged**